tests/Fixer/Whitespace/BlankLineBetweenImportGroupsFixerTest.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Whitespace;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Whitespace\BlankLineBetweenImportGroupsFixer
+ */
+final class BlankLineBetweenImportGroupsFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideFixTypesOrderAndWhitespaceCases
+     */
+    public function testFixTypesOrderAndNewlines(string $expected, ?string $input = null): void
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixTypesOrderAndWhitespaceCases(): array
```

Comment:
I think we need some tests with some funky comments too, to make sure we don't accidentality move some code inside a comment.

Reply:
I have added some test cases, and I hope they're funky enough with the comments. Though I am not entirely sure what the actual expected behavior ought to be, as the PSR12 doesn't allow comments within the use blocks if I recall correctly.

So if you have any more test cases, please do let me know. I hope the current tests are sufficient.

Reply:
I don't think you have any tests with comments inside `use` statements:

```php
use /* x */ function /* x */ Psl\Str\trim;
use /* x */ Psl\Str /* x */;
use /* x */ const /* x */ Psl\Str\ALPHABET;
use /* x */ function /* x */ Psl\Str\ /* x */ {
    /* x */ trim_left /* x */,
    /* x */ trim_right /* x */,
};
```

Reply:
Wow, I've just discovered that you can even do this (it imports `Psl\Str\trim_left`):

```php
use function P/**/s/**/l/**/\/**/S/**/t/**/r/**/\/**/t/**/r/**/i/**/m/**/_/**/l/**/e/**/f/**/t/**/;
```

Probably no need to test that in this fixer though :)

Reply:
I've added the tests for the initial tests with the comments inside the use statement. In PHP 7, the following block results in a syntax error, so I couldn't perform these.

```php
use function P/**/s/**/l/**/\/**/S/**/t/**/r/**/\/**/t/**/r/**/i/**/m/**/_/**/l/**/e/**/f/**/t/**/;
```